react/react.d.ts

```diff
@@ -122,12 +121,12 @@ declare namespace __React {
     type ReactInstance = Component<any, any> | Element;
 
     // Base component for plain JS classes
-    class Component<P, S> implements ComponentLifecycle<P, S> {
+    abstract class Component<P, S> implements ComponentLifecycle<P, S> {
         constructor(props?: P, context?: any);
         setState(f: (prevState: S, props: P) => S, callback?: () => any): void;
         setState(state: S, callback?: () => any): void;
         forceUpdate(callBack?: () => any): void;
-        render(): JSX.Element;
+        abstract render(): JSX.Element;
```

Comment:
A number of the failures from this were from the ``antd`` typings. I've submitted a fix for that in #8561 so that set of failures would go away. (Someone that uses ``material-ui`` probably wouldn't find it hard to update those typings...)